feature/storedetail/src/main/java/com/hankki/feature/storedetail/StoreDetailRoute.kt

```diff
@@ -34,40 +32,63 @@ import androidx.compose.ui.unit.dp
 import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import com.hankki.core.common.extension.noRippleClickable
-import com.hankki.core.designsystem.theme.HankkiTheme
-import com.hankki.core.designsystem.theme.HankkijogboTheme
+import com.hankki.core.common.utill.UiState
 import com.hankki.core.designsystem.R
 import com.hankki.core.designsystem.component.button.HankkiButton
 import com.hankki.core.designsystem.component.button.StoreDetailButton
-import com.hankki.core.designsystem.component.layout.StoreDetailMenuBox
 import com.hankki.core.designsystem.component.topappbar.HankkiTopBar
+import com.hankki.core.designsystem.theme.Gray400
 import com.hankki.core.designsystem.theme.Gray900
+import com.hankki.core.designsystem.theme.HankkiTheme
+import com.hankki.core.designsystem.theme.HankkijogboTheme
+import com.hankki.feature.storedetail.component.StoreDetailMenuBox
 import com.hankki.feature.storedetail.model.MenuItem
-import com.hankki.feature.storedetail.model.StoreDetail
+import kotlinx.collections.immutable.PersistentList
+import kotlinx.collections.immutable.toPersistentList
 
 @Composable
 fun StoreDetailRoute() {
     val viewModel: StoreDetailViewModel = hiltViewModel()
-    val storeDetail by viewModel.storeDetail.collectAsStateWithLifecycle()
+    val storeDetailState by viewModel.storeDetail.collectAsStateWithLifecycle()
+    val isLiked by viewModel.isLiked.collectAsStateWithLifecycle()
+    val heartCount by viewModel.heartCount.collectAsStateWithLifecycle()
+    val selectedIndex by viewModel.selectedIndex.collectAsStateWithLifecycle()
+    val buttonLabels by viewModel.buttonLabels.collectAsStateWithLifecycle()
 
-    storeDetail?.let {
-        StoreDetailScreen(
-            storeDetail = it,
-            onLikeClicked = { viewModel.toggleLike() }
-        )
-    }
-}
+    when (val state = storeDetailState) {
+        is UiState.Loading -> {}
 
-@Composable
-fun StoreDetailScreen(storeDetail: StoreDetail, onLikeClicked: () -> Unit) {
-    var isLiked by remember { mutableStateOf(storeDetail.isLiked) }
-    var heartCount by remember { mutableIntStateOf(storeDetail.heartCount) }
+        is UiState.Success -> {
+            val storeDetail = state.data
+            StoreDetailScreen(
+                title = storeDetail.name,
+                tag = storeDetail.category,
+                menuItems = storeDetail.menus.toPersistentList(),
+                isLiked = isLiked,
+                heartCount = heartCount,
+                selectedIndex = selectedIndex,
+                buttonLabels = buttonLabels,
+                onLikeClicked = { viewModel.toggleLike() },
+                onSelectIndex = { index -> viewModel.updateSelectedIndex(index) }
+            )
```

Comment:
객체를 생성하지 말고 아래와 같은 방법은 어떨까요?
```suggestion
        with(state.data){
            StoreDetailScreen(
                title = name,
                tag = category,
                menuItems = menus.toPersistentList(),
                isLiked = isLiked,
                heartCount = heartCount,
                selectedIndex = selectedIndex,
                buttonLabels = buttonLabels,
                onLikeClicked = { viewModel.toggleLike() },
                onSelectIndex = { index -> viewModel.updateSelectedIndex(index) }
            )
        }
```